server/src/test/java/com/microsoft/java/bs/core/internal/server/IntegrationTest.java

```diff
@@ -0,0 +1,431 @@
+    protected final List<ShowMessageParams> showMessages = new ArrayList<>();
+    protected final List<TestReport> testReports = new ArrayList<>();
+    protected final List<TestStartEx> testStarts = new ArrayList<>();
+    protected final List<TestFinishEx> testFinishes = new ArrayList<>();
+
+    protected void clearMessages() {
+      startReports.clear();
+      finishReports.clear();
+      compileReports.clear();
+      compileTasks.clear();
+      logMessages.clear();
+      showMessages.clear();
+      testReports.clear();
+      testStarts.clear();
+      testFinishes.clear();
+    }
+
+    protected void waitOnStartReports(int size) {
+      waitOnMessages("Start Reports", size, startReports::size);
+    }
+
+    protected void waitOnFinishReports(int size) {
+      waitOnMessages("Finish Reports", size, finishReports::size);
+    }
+
+    protected void waitOnCompileReports(int size) {
+      waitOnMessages("Compile Reports", size, compileReports::size);
+    }
+
+    protected void waitOnCompileTasks(int size) {
+      waitOnMessages("Compile Tasks", size, compileTasks::size);
+    }
+
+    protected void waitOnLogMessages(int size) {
+      waitOnMessages("Log Messages", size, logMessages::size);
+    }
+
+    protected void waitOnShowMessages(int size) {
+      waitOnMessages("Show Messages", size, showMessages::size);
+    }
+
+    protected void waitOnTestReports(int size) {
+      waitOnMessages("Test Reports", size, testReports::size);
+    }
+
+    protected void waitOnTestStarts(int size) {
+      waitOnMessages("Test Starts", size, testStarts::size);
+    }
+
+    protected void waitOnTestFinishes(int size) {
+      waitOnMessages("Test Finishes", size, testFinishes::size);
+    }
+
+    protected long finishReportErrorCount() {
+      return finishReports.stream()
+          .filter(report -> report.getStatus() == StatusCode.ERROR)
+          .count();
+    }
+
+    private static List<String> getTestNameHierarchy(TestName testName) {
+      List<String> names = new LinkedList<>();
+      while (testName != null) {
+        names.add(testName.getDisplayName());
+        testName = testName.getParent();
+      }
+      return names;
+    }
+
+    private static boolean matchesTest(TestName testName, String suiteName, String className,
+                                       String methodName, List<String> testNames) {
+      return Objects.equals(testName.getSuiteName(), suiteName)
+          && Objects.equals(testName.getClassName(), className)
+          && Objects.equals(testName.getMethodName(), methodName)
+          && Objects.equals(getTestNameHierarchy(testName), testNames);
+    }
+
+    private static String testNameAsString(TestName testName) {
+      return testName.getSuiteName() + "," + testName.getClassName() + ","
+          + testName.getMethodName() + "," + getTestNameHierarchy(testName);
+    }
+
+    TestStartEx getTestStart(String suiteName, String className, String methodName,
```

Comment:
Since this is the base class, I think it should only provide the minimum required members. These test related method better move to the `BuildTargetServiceIntegrationTest`?

Reply:
Fixed in https://github.com/microsoft/build-server-for-gradle/pull/165/commits/7e4749a704b01421c5b6282ea0204d8856b32636

Reply:
Fixed in https://github.com/microsoft/build-server-for-gradle/pull/165/commits/7e4749a704b01421c5b6282ea0204d8856b32636